custom_components/pun_sensor/sensor.py

```diff
@@ -230,37 +229,42 @@ def options(self) -> list[str] | None:
         return ["F1", "F2", "F3"]
 
     @property
+
     def native_value(self) -> str | None:
         """Restituisce la fascia corrente come stato"""
-        return decode_fascia(self.coordinator.fascia_corrente)
+        if not self.coordinator.fascia_corrente:
+            return "None"
+        return self.coordinator.fascia_corrente.value
 
-    @override
     @property
     def extra_state_attributes(self) -> dict[str, Any] | None:
         return {
-            'fascia_successiva': decode_fascia(self.coordinator.fascia_successiva),
-            'inizio_fascia_successiva': self.coordinator.prossimo_cambio_fascia,
-            'termine_fascia_successiva': self.coordinator.termine_prossima_fascia
+            "fascia_successiva": self.coordinator.fascia_successiva.value
+            if self.coordinator.fascia_successiva
+            else "None",
```

Comment:
Non dovrebbe essere `None` (senza virgolette) come nella vecchia funzione `decode_fascia` e come sepcifica anche il `def`?

Reply:
eh anche qua piu' o meno, questo restituisce direttamente 3 valori, quindi in realta' e' sbagliata la definizione, che era rimasta da quando c'era la decode fascia, in teoria ora ritorna un dict di stringhe, pero' si dovrei poter mettere None, devo testare un attimo non ricordo bene perche' avevo messo le virgolette 😅